crates/runtime/src/model/tool_use.rs

```diff
@@ -198,7 +199,7 @@ impl ToolUsingChat {
                 .into();
 
         let mut tool_and_response_content = vec![];
-        for t in spiced_tools {
+        for t in spiced_tools.clone() {
```

Comment:
doesn't need cloning?

Reply:
It is accessed below to filter out `list_datasets`, so have to clone